tests/python/unittest/test_operator.py

```diff
@@ -367,6 +367,108 @@ def test_pow_fn():
     check_symbolic_forward(y, [x], [2**x])
     check_symbolic_backward(y, [x], [np.ones(shape)], [np.log(2) * 2**x])
 
+def test_equal():
```

Comment:
If nothing else is missing, can I suggest to handle this as a separate issue?
It would be easier for me to make a one file change separately instead of carrying the rest of the files around. Let me know I can create a separate issue and create a separate PR for it.